src/dimension.c

```diff
@@ -516,6 +516,13 @@ dimension_insert(int32 hypertable_id,
 int
 dimension_set_type(Dimension *dim, Oid newtype)
 {
+	if (!IS_VALID_OPEN_DIM_TYPE(newtype))
+		ereport(ERROR,
+				(errcode(ERRCODE_DATATYPE_MISMATCH),
+				 errmsg("time dimension column of a hypertable cannot be converted to "
```

Comment:
PG throws this error when changing data types on views:

```
	ereport(ERROR,
					(errcode(ERRCODE_INVALID_TABLE_DEFINITION),
					 errmsg("cannot change data type of view column \"%s\" from %s to %s",
							NameStr(oldattr->attname),
							format_type_with_typemod(oldattr->atttypid,
													 oldattr->atttypmod),
							format_type_with_typemod(newattr->atttypid,
													 newattr->atttypmod))));
```

I'd format this message very similar to the above, with same error code. Basically change `view` to `hypertable`.

Reply:
You can add a `errdetail` or `errhint` with more information specific to hypertables if that makes sense.

Reply:
ok, changed. Some error messages in timescale directly print OID of an object/type and are not using format functions. 
I think I also am not using it in `show_chunks`. Shall I add it there too?